app/models/category.rb

```diff
@@ -0,0 +1,4 @@
+class Category < ApplicationRecord
+  has_many :products
+  has_many :banners
```

Comment:
I think this relationship is backwards, isn't it?

A banner has many categories, and a category belongs to a banner